tutorials/intro/plot_introduction.py

```diff
@@ -1,404 +1,293 @@
-# file with::
+###############################################################################
+# Loading data
+# ^^^^^^^^^^^^
 #
-#     >>> mne.set_config('MNE_LOGGING_LEVEL', 'WARNING')
+# MNE-Python data structures are based around the FIF file format from
+# Neuromag, but there are reader functions for :ref:`a wide variety of other
+# data formats <data-formats>`. MNE-Python also has interfaces to a
+# variety of :doc:`publicly available datasets <../../manual/datasets_index>`,
+# which MNE-Python can download and manage for you.
 #
-# Note that the location of the mne-python preferences file (for easier manual
-# editing) can be found using:
-
-print(mne.get_config_path())
-
-##############################################################################
-# By default logging messages print to the console, but look at
-# :func:`mne.set_log_file` to save output to a file.
+# We'll start this tutorial by loading one of the example datasets (called
+# ":ref:`sample-dataset`"), which contains EEG and MEG data from one subject
+# performing an audiovisual experiment, along with structural MRI scans for
+# that subject. The :func:`mne.datasets.sample.data_path` function will
+# automatically download the dataset if it isn't found in one of the expected
+# locations, then return the directory path to the dataset (see the
+# documentation of :func:`~mne.datasets.sample.data_path` for a list of places
+# it checks before downloading).
+
+sample_data_folder = mne.datasets.sample.data_path()
+sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
+                                    'sample_audvis_raw.fif')
+raw = mne.io.read_raw_fif(sample_data_raw_file, preload=False)
+
+###############################################################################
+# .. note::
 #
-# Access raw data
-# ^^^^^^^^^^^^^^^
-
-from mne.datasets import sample  # noqa
-data_path = sample.data_path()
-raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
-print(raw_fname)
-
-##############################################################################
-# .. note:: The MNE sample dataset should be downloaded automatically but be
-#           patient (approx. 2GB)
+#     :func:`~mne.io.read_raw_fif` takes a ``preload`` parameter, which
+#     determines whether the data will be copied into RAM or not. Some
+#     operations (such as filtering) require that the data be preloaded, but it
+#     is possible use ``preload=False`` and then copy raw data into memory
+#     later using the :meth:`~mne.io.Raw.load_data` method if needed.
+#
+# By default, :func:`~mne.io.read_raw_fif` displays some information about the
+# file it's loading; for example, here it tells us that there are three
+# "projection items" in the file along with the recorded data; those are
+# :term:`SSP projectors <projector>` for environmental noise, and are discussed
+# in a later tutorial. In addition to the information displayed during loading,
```

Comment:
SSP is a very MEG centric concept and I fear you would alienate the EEG audience with this. Perhaps better to show with ICA?

Reply:
I'd say it's fine to mention SSP projectors here, though agree with EEG point, see next comment below...

Reply:
@jasmainak I was avoiding ICA because I think it takes too long to execute/explain, and you kind of need to show a lot of figures.  Here's what I think it would take for ICA (replaces 4 lines of code and 2 comments in the SSP version:

```python
# set up and fit the ICA; avoid fitting to huge (presumably environmental)
# artifacts by ignoring signal exceeding 5000 fT; 4000 fT/cm; 250 μV
reject_criteria = dict(mag=5e-12, grad=4e-10, eeg=250e-6)
ica = mne.preprocessing.ICA(n_components=20, random_state=97, max_iter=800)
ica.fit(raw, reject=reject_criteria)
# detect suspected EOG events
eog_epochs = mne.preprocessing.create_eog_epochs(raw, reject=reject_criteria)
eog_average = eog_epochs.copy().average()
# see which component(s) it thinks are capturing EOG
eog_inds, eog_scores = ica.find_bads_eog(eog_epochs)
ica.plot_scores(eog_scores, exclude=eog_inds)
ica.plot_sources(eog_average, exclude=eog_inds)
# check that removing the suspected component(s) improves the signal
ica.plot_overlay(eog_average, exclude=eog_inds)

# now repeat the process for ECG (we'll skip the plotting steps)
ecg_epochs = mne.preprocessing.create_ecg_epochs(raw)
ecg_inds, ecg_scores = ica.find_bads_ecg(ecg_epochs, method='ctps')

###############################################################################
# Once we're confident about which component(s) we want to remove, we add those
# to the "exclude" list and then apply the ICA to the raw signal. The
# :meth:`~mne.preprocessing.ICA.apply` method requires the raw data to be
# loaded into memory (by default it's read from disk as-needed), so we'll use
# :meth:`~mne.io.Raw.load_data` first. We'll also make a copy of the
# :class:`~mne.io.Raw` object so we can compare the signal before and after
# artifact removal side-by-side:

orig_raw = raw.copy()
raw.load_data()
ica.exclude.extend(eog_inds + ecg_inds)
ica.apply(raw)

orig_raw.plot(group_by='position', start=12, duration=4)
raw.plot(group_by='position', start=12, duration=4)
```

Reply:
Hmm ... this would be the case if you wanted automatic ICA component detection. The four liner in my mind was:
```py
ica = ICA(n_components=25, method='fastica', random_state=23).fit(raw)
exclude = [1]  # say the components corresponding to eyeblink and heart beat
ica.plot_properties(picks=exclude)
raw_clean = ica.apply(raw.copy())
```

Reply:
I’m with @drammock on this one. Would it help to write a note to the effect of:
‘SSPs are spatial filters calculated using PCA on identified artifacts such as eye blinks’
I’m not too keen on using the term ‘spatial filter’, but I believe the EEG community is used to this form of processing, they just refer to it differently (PCA vs SSP, same difference, more or less)

Reply:
I actually ended up including a shortened version of ICA (in part because it does a slightly better job, resulting in nicer STC plots at the end).  It's not as succinct as @jasmainak's 4-liner, but I think it's probably worth it to have a preprocessing example that's more familiar to a wider audience.